### PR TITLE
ddl: fix bug when add index meet math.MaxInt64 handle.

### PR DESCRIPTION
**Pull request by winkyao**
Before this PR, we will skip adding index when handle == math.MaxInt64. We can use below ops to reproduce this bug in master:

```
mysql> create table t(a bigint PRIMARY KEY, b int);
Query OK, 0 rows affected (0.10 sec)

mysql> insert into t values(9223372036854775807, 1);
Query OK, 1 row affected (0.00 sec)

mysql> alter table t add index idx_b(b);
Query OK, 0 rows affected (0.03 sec)

mysql> admin check table t;
ERROR 1105 (HY000): t err:[admin:1]index:<nil> != record:&{9223372036854775807 [{1 0 0 0 1 [] <nil>}]}
```

### CODE REVIEW

**Comment by winkyao** [comment]
@zimulala @coocood PTAL

**Comment by coocood** [comment]
LGTM

File: ddl/index.go [hunk 1]
```diff
@@ -432,8 +432,9 @@ type addIndexWorker struct {
 }
 
 type reorgIndexTask struct {
-	startHandle int64
-	endHandle   int64
+	startHandle  int64
+	endHandle    int64
+	endInclusive bool
```
Comment (zhexuany) [inline ddl/index.go]: endHandleInclused maybe better？

**Comment by shenli** [comment]
@zhexuany PTAL

**Comment by zhexuany** [comment]
LGTM

File: ddl/index.go [hunk 2]
```diff
@@ -434,6 +434,7 @@ type addIndexWorker struct {
 type reorgIndexTask struct {
 	startHandle int64
 	endHandle   int64
+	endIncluded bool
```
Comment (shenli) [inline ddl/index.go]: Add a comment for this. Why we need it?

File: ddl/index.go [hunk 3]
```diff
@@ -524,6 +529,10 @@ func (w *addIndexWorker) fetchRowColVals(txn kv.Transaction, taskRange reorgInde
 			}
 
 			w.idxRecords = append(w.idxRecords, idxRecord)
+			if handle == taskRange.endHandle {
```
Comment (shenli) [inline ddl/index.go]: Should we consider `taskRange.endIncluded` here?

Comment (tiancaiamao) [inline ddl/index.go]: Is this check necessary here?

Comment (zimulala) [reply ddl/index.go]: Is It checked in line 516？

Comment (winkyao) [reply ddl/index.go]: If !taskRange.endIncluded, we will not reach here when handle == taskRange.endHandle , we return in 525

Comment (winkyao) [reply ddl/index.go]: yes,, we need handleOutOfRange be true, outside will use this boolean.

**Comment by winkyao** [comment]
@zimulala @tiancaiamao PTAL

File: ddl/index.go [hunk 4]
```diff
@@ -512,8 +515,12 @@ func (w *addIndexWorker) fetchRowColVals(txn kv.Transaction, taskRange reorgInde
 	handleOutOfRange := false
 	err := iterateSnapshotRows(w.sessCtx.GetStore(), w.table, txn.StartTS(), taskRange.startHandle,
 		func(handle int64, recordKey kv.Key, rawRow []byte) (bool, error) {
-			// Don't create endHandle index.
-			handleOutOfRange = handle >= taskRange.endHandle
+			if !taskRange.endIncluded {
+				handleOutOfRange = handle >= taskRange.endHandle
+			} else {
+				handleOutOfRange = handle > taskRange.endHandle
+			}
```
Comment (zimulala) [inline ddl/index.go]: Could we use `handleOutOfRange = handle > taskRange.endHandle` instead of line518-522?

Comment (shenli) [reply ddl/index.go]: @zimulala No, line 519 has different condition with line 521.

File: ddl/index.go (hunk 3 above)
Comment (shenli) [reply ddl/index.go]: Could comment for this?

File: ddl/index.go (hunk 4 above)
Comment (zimulala) [reply ddl/index.go]: @shenli But I think line 535 done the same thing as line 519.

Comment (winkyao) [reply ddl/index.go]: @zimulala You can run all the tests because we will never add the same index twice.

File: ddl/index.go (hunk 3 above)
Comment (winkyao) [reply ddl/index.go]: Done.

**Pull request merged**